.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+  return github.rest.issues.create({
+    owner: context.payload.repository.owner.login,
+    repo: "opentelemetry-collector-contrib",
+    title,
+    body: report,
+    labels: ["report"]
+  })
+}
+
+async function getLastWeeksReport({ github, since, context }) {
+  const issues = await github.rest.issues.listForRepo({
+    owner: context.payload.repository.owner.login,
+    repo: 'opentelemetry-collector-contrib',
+    state: 'all', // To get both open and closed issues
+    labels: ["report"],
+    since: since.toISOString(),
+    per_page: 1,
+    direction: "asc"
+  });
+  if (issues.data.length === 0) {
+    return null;
+  } 
+  // grab earliest issue if multiple
+  return issues.data[0];
+}
+
+function parseJsonFromText(text) {
+  // Use regex to find the JSON data within the <pre></pre> tags
+  const regex = /<pre>\s*([\s\S]*?)\s*<\/pre>/;
+  const match = text.match(regex);
+
+  if (match && match[1]) {
+    // Parse the found string to JSON
+    return JSON.parse(match[1]);
+  } else {
+    throw new Error("JSON data not found");
+  }
+}
+
+async function processIssues({ github, context, lookbackData }) {
+  const issuesData = await getIssuesData({github, context});
+
+  const prevReportLookback = new Date(lookbackData.sevenDaysAgo)
+  prevReportLookback.setDate(prevReportLookback.getDate() - 7)
+  const previousReportIssue = await getLastWeeksReport({github, since: prevReportLookback, context});
+  // initialize to zeros
+  let previousReport = null;
+
+  if (previousReportIssue !== null) {
+    const {created_at, id, url, title} = previousReportIssue;
+    debug({ msg: "previous issue", created_at, id, url, title })
+    previousReport = parseJsonFromText(previousReportIssue.body)
+  }
+
+  return {issuesData, previousReport}
+
+
+}
+
+const findFilesByName = (startPath, filter) => {
+  let results = [];
+
+  // Check if directory exists
+  let files;
+  try {
+      files = fs.readdirSync(startPath);
+  } catch (error) {
+      console.error("Error reading directory: ", startPath, error);
+      return [];
+  }
+
+  for (let i = 0; i < files.length; i++) {
+      const filename = path.join(startPath, files[i]);
+      let stat;
+      try {
+          stat = fs.lstatSync(filename);
+      } catch (error) {
+          console.error("Error stating file: ", filename, error);
+          continue;
+      }
+      
+      if (stat.isDirectory()) {
+          const innerResults = findFilesByName(filename, filter); // Recursive call
+          results = results.concat(innerResults);
+      } else if (filename.indexOf(filter) >= 0) {
+          results.push(filename);
+      }
+  }
+  return results;
+};
+
+function processFiles(files) {
+  const results = {};
+
+  for (let filePath of files) {
+      const component = path.basename(path.dirname(path.dirname(filePath)));  // Top level directory
```

Comment:
If I understand correctly, by `component` you mean `component class` - a receiver, or a processor, etc. Reading the component type from the path should work, but a cleaner way would be to use the `status.class` field from the metadata file contents.

Reply:
this is because we currently collect `metadata.yaml` from **everything**. this includes internal components that don't have a `status.class` field (eg. processor/resourcedetectionprocessor/internal/aws/ec2/metadata.yaml). though to be fair. added a check for the `status.class` to set component and default to the current method if nothing is found